SIPS/sip-23.md

```diff
@@ -0,0 +1,627 @@
+
+```typescript
+type SnapNode =
+  | SnapElement
+  | string
+  | null
+  | SnapNode[];
+```
+
+### Interface structure
+
+The Snap interface structure is defined using JSX components, and consists of:
+
+- A container element, `Container`, which wraps the entire interface.
+- A box element, `Box`, which contains the main content of the interface.
+- An optional footer element, `Footer`, which appears at the bottom of the
+  interface.
+
+![Snap interface structure](../assets/sip-23/interface-structure.png)
+
+#### Example
+
+Below is an example of a simple Snap interface structure using JSX components:
+
+```typescript jsx
+<Container>
+  <Box>
+    <Heading>My Snap</Heading>
+    <Text>Hello, world!</Text>
+  </Box>
+  <Footer>
+    <Button onClick={handleClick}>Click me</Button>
+  </Footer>
+</Container>
+```
+
+Each of the components used in the example is defined in the following sections.
+
+### Components
+
+#### Address
+
+The `Address` component renders a text element with an address. It accepts an
+`address` prop which is a string representing an Ethereum address.
+
+```typescript
+type AddressProps = {
+  address: `0x${string}`;
+};
+
+const Address: SnapComponent<AddressProps>;
+```
+
+##### Example
+
+```typescript jsx
+<Address address="0x4bbeEB066eD09B7AEd07bF39EEe0460DFa261520" />
+```
+
+#### Bold
+
+The `Bold` component renders a text element in bold. It accepts a `children`
+prop which can be a `string` or an array of `string`s. The component MUST be
+used inside a `Text` component.
+
+```typescript
+type BoldProps = {
+  children: string | string[];
+};
+
+const Bold: SnapComponent<BoldProps>;
+```
+
+##### Example
+
+```typescript jsx
+<Text>
+  Hello, <Bold>world</Bold>!
+</Text>
+```
+
+#### Box
+
+The `Box` component renders a box element. It accepts a `children` prop which
+can be a `SnapNode`. The component is used to group elements together.
+
+In addition, the component accepts an optional `alignment` prop which can be
+`'start'`, `'center'`, `'end'`, or `undefined`. The default value is
+`undefined`.
+
+```typescript
+type BoxProps = {
+  children: SnapNode;
+  alignment?: 'start' | 'center' | 'end' | undefined;
+};
+
+const Box: SnapComponent<BoxProps>;
+```
+
+##### Example
+
+```typescript jsx
+<Box alignment="center">
+  <Text>Hello, world!</Text>
+</Box>
+```
+
+#### Button
+
+The `Button` component renders a button element. It accepts a `children` prop
+which can be a `string` or an array of `string`s. The component also accepts
+optional `type`, `variant`, and `onClick` props. The `type` prop MUST be one of
+`'button'`, `'submit'`, or `undefined`. The `variant` prop MUST be one of
+`'primary'`, `'secondary'`, or `undefined`. The `onClick` prop is a function to
+be called when the button is clicked.
+
+```typescript
+type ButtonProps = {
+  children: string | string[];
+  type?: 'button' | 'submit' | undefined;
+  variant?: 'primary' | 'secondary' | undefined;
```

Comment:
Wondering if we want to change this. I believe Erik N has thoughts on removing some of the variants